### PR TITLE
[JENKINS-26693] Workflow support

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-26693](https://issues.jenkins-ci.org/browse/JENKINS-26693), superseding #5.

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java [hunk 1]
```diff
@@ -68,22 +72,28 @@
...
-        AbstractProject<?,?> p = project(job, req, rsp);
-        ParametersDefinitionProperty pp = p.getProperty(ParametersDefinitionProperty.class);
+        ParameterizedJobMixIn.ParameterizedJob p = project(job, req, rsp);
+        if (delay == null) {
+            delay = new TimeDuration(p.getQuietPeriod());
+        }
+        ParametersDefinitionProperty pp = ((Job<?,?>) p).getProperty(ParametersDefinitionProperty.class);
         if (pp == null) {
             LOGGER.fine("wrong kind");
             throw HttpResponses.error(HttpServletResponse.SC_BAD_REQUEST, "use build for this build");
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java]: This message (and the corresponding one in `build(…)` could be improved to say something like `Use the /build URL to build this job`.

File: src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java [hunk 2]
```diff
@@ -95,29 +105,46 @@ public void doBuildWithParameters(StaplerRequest req, StaplerResponse rsp, @Quer
...
 
     public void doPolling(StaplerRequest req, StaplerResponse rsp, @QueryParameter String job) throws IOException, ServletException {
         LOGGER.log(Level.FINE, "polling on {0}", job);
-        project(job, req, rsp).schedulePolling();
+        ParameterizedJobMixIn.ParameterizedJob p = project(job, req, rsp);
+        // AbstractProject.schedulePolling only adds one thing here: check for isDisabled. But in that case, !isBuildable, so we would not have gotten here anyway.
+        SCMTriggerItem scmp = SCMTriggerItem.SCMTriggerItems.asSCMTriggerItem(p);
+        if (scmp == null) {
+            LOGGER.log(Level.FINE, "{0} is not a SCMTriggerItem", p);
+            throw HttpResponses.error(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, new IOException(job + " is not a SCMTriggerItem"));
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java]: Why are these internal server errors and the ones in `doBuild`/`doBuildWithParameters` are bad request? All of these only depend on the configuration of the project selected as parameter.

File: src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java (hunk 1 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java]: Yes, it could, though that would better be left to a separate PR; file a JIRA ticket for it.

File: src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java (hunk 2 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java]: I think you are right, a 400 would be more appropriate. The HTTP spec is kind of vague about what it means.

**Comment by stephenc** [comment]
:+1:
